src/DonorDashboards/Repositories/Donations.php

```diff
@@ -83,20 +155,42 @@ public function getAverageRevenue($donorId)
      */
     private function getDonationAggregate($rawAggregate, $donorId)
     {
+        return $this->queryDonationRevenueData($donorId, "$rawAggregate as result");
+    }
+
+    /**
+     * Query donation revenue data from the revenue table for a specific donor
+     *
+     * @unreleased
+     *
+     * @param int    $donorId
+     * @param string $selectClause
+     *
+     * @return object|null
+     */
+    private function queryDonationRevenueData($donorId, $selectClause)
```

Comment:
The idea of passing SQL as a parameter makes me anxious 😅
I'd suggest refactoring this method to something like `getRevenueDonationDataQuery`, which will return a `QueryBuilder` object with a base query. Then, when you call it, it will look something like this:
```
$query = $this->getRevenueDonationDataQuery($ids);
$query->select('whatever');
$data = $query->getAll();
```
If you are not familiar with our QueryBuilder library, you can check it here  https://github.com/impress-org/givewp/blob/develop/src/Framework/QueryBuilder/README.md